app/services/asset_derivative_storage_type_auditor.rb

```diff
@@ -79,13 +98,47 @@ def notify_if_failed
 
   # Checks, records, and notifies of failures
   def perform!
+    log_start
     check_all
-    #store_audit_results
+    store_audit_results
     notify_if_failed
+    delete_stale_reports
+    log_end
   end
 
   private
 
+  # Get or create the report
+  def report
+    @report ||= Admin::AssetDerivativeStorageTypeReport.create!
+  end
+
+  def log_start
+    log_into_report({ start_time: Time.now.to_s })
+  end
+
+  def log_end
+    log_into_report({ end_time:   Time.now.to_s })
+  end
+
+  def log_into_report(data)
+    report.data_for_report.update(data)
+    report.save!
+  end
+
+  def delete_stale_reports
+    sql = """
+      SELECT old.id FROM asset_derivative_storage_type_reports old
+      WHERE old.id NOT IN (
+        SELECT id FROM asset_derivative_storage_type_reports new
+        ORDER BY created_at DESC
+        LIMIT 1 )
+    """
```

Comment:
You could probably do this with ActiveRecord instead of raw SQL, which might be more reliable with regard to future code changes and avoiding other kinds of bugs. But this is probably good enough for now, why not. 

Although wait we already know what report we created. How about simply:  `AssetDerivativeStorageTypeReports.where.not(id: report.id).delete_all` (or `destroy_all` if you wanted to trigger Rails callbacks, which shoudln't probably matter here).

Reply:
You are so, so right about this. Thanks :)